R/08-create-final-reports.R

```diff
@@ -0,0 +1,429 @@
+#   - is_internal: whether this is the full internal dataset or not (has extra columns to format)
+# outputs:
+#   - none. Saves a copy of the data to the LAN at the specified location
+create_final_excel <- function(
+  template, 
+  final_excel,
+  grad_data,
+  occ_data,
+  is_draft=TRUE,
+  is_internal=TRUE
+){
+  # load template 
+  outwb <- loadWorkbook(template)
+  
+  # if draft, add 'DRAFT' to first page 
+  if (is_draft){
+    writeData(outwb, "User Guide", x='DRAFT', startRow=1, startCol=1)
+    addStyle(outwb, "User Guide", style = csDraft, rows = 1, cols = 1)
+  } 
+  
+  ## add new sheet for grads 
+  sheet <- addWorksheet(outwb, sheetName="Graduate Projections") 
+  n_rows <- nrow(grad_data)
+  n_cols <- length(grad_data)
+  
+  # add data to sheet
+  startRow <- 1
+  if (is_draft){
+    writeData(outwb, sheet, x='DRAFT', startRow=1, startCol=1)
+    addStyle(outwb, sheet, style = csDraft, rows = 1, cols = 1)
+    startRow <- 2
+  } 
+  
+  writeData(outwb, sheet, grad_data, colNames = TRUE, rowNames = FALSE, startRow=startRow, startCol=1, withFilter = FALSE,
+            keepNA = FALSE)
+  
+  # Freeze top row
+  freezePane(outwb,sheet, firstActiveRow=startRow+1)
+  
+  # style headers
+  addStyle(outwb, sheet, style=csRegularBold, rows=startRow, cols=1:n_cols)
+  
+  # set col widths
+  cred_col <- which(names(grad_data) == "Credential Type")
+  setColWidths(outwb,sheet,cols = cred_col, widths = "auto")
+  
+  ## add new sheet for occupations 
+  sheet <- addWorksheet(outwb, sheetName="Occupation Projections") 
+  n_rows <- nrow(occ_data)
+  n_cols <- length(occ_data)
+  
+  # add data to sheet 
+  startRow <- 1
+  if (is_draft){
+    writeData(outwb, sheet, x='DRAFT', startRow=1, startCol=1)
+    addStyle(outwb, sheet, style = csDraft, rows = 1, cols = 1)
+    startRow <- 2
+  } 
+  
+  writeData(
+    outwb, sheet, occ_data, 
+    colNames = TRUE, rowNames = FALSE, 
+    startRow=startRow, startCol=1, 
+    withFilter = TRUE,
+    keepNA = FALSE
+    )
+  
+  # Freeze top row
+  freezePane(outwb, sheet, firstActiveRow=startRow+1)
+  
+  # style headers
+  addStyle(outwb, sheet, style=csRegularBold, rows=startRow, cols=1:n_cols)
+  setRowHeights(outwb, sheet, rows=startRow, heights=60)
+  
+  # set col widths
+  occ_col <- which(names(occ_data) == "Occupation Description")
+  setColWidths(outwb,sheet,cols = occ_col, widths = 40)
+  
+  # extra internal release columns
+  if (is_internal){
+    rg_col <- which(names(occ_data) == "Region Name")
+    setColWidths(outwb,sheet,cols = rg_col, widths = "auto")
+    # style the percentages 
+    qi_col <- which(names(occ_data) == "Quality Indicator")
+    ci_col <- which(names(occ_data) =="Public Post-Secondary Coverage Indicator")
+    addStyle(outwb, sheet, style = csPerc, rows=(startRow+1):(n_rows+2), cols=qi_col)
+    addStyle(outwb, sheet, style = csPerc, rows=(startRow+1):(n_rows+2), cols=ci_col)
```

Comment:
I think the 2 here is for draft mode technically, or otherwise it would just be 1 (in case we wanted to do shading or something this might matter)
```suggestion
    addStyle(outwb, sheet, style = csPerc, rows=(startRow+1):(n_rows+startRow), cols=qi_col)
    addStyle(outwb, sheet, style = csPerc, rows=(startRow+1):(n_rows+startRow), cols=ci_col)
```